tests/FSharp.Compiler.ComponentTests/Interop/StaticsInInterfaces.fs

```diff
@@ -68,4 +68,127 @@ let main _ =
+            ...
+        }
+    *)
+    #if !NETCOREAPP
+    [<Fact(Skip = "NET472 is unsupported runtime for this kind of test.")>]
+    #else
+    [<Fact>]
+    #endif
+    let ``F# generates valid IL for abstract static interface methods`` () =
+
+        let csharpLib = csharpBaseClass 
+
+        let fsharpSource =
+            """
+module StaticsTesting 
+open StaticsInInterfaces
+
+type MyRepeatSequence() =
+    interface IGetNext<MyRepeatSequence> with
+        static member Next(other: MyRepeatSequence) : MyRepeatSequence = other 
+"""
+        Fsx fsharpSource
+        |> withLangVersionPreview
+        |> withReferences [csharpLib]
+        |> compile
+        |> shouldSucceed
+        |> verifyIL [
+        """
+.class public abstract auto ansi sealed StaticsTesting
+extends [runtime]System.Object
+{
+  .custom instance void [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::.ctor(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags) = ( 01 00 07 00 00 00 00 00 ) 
+  .class auto ansi serializable nested public MyRepeatSequence
+  extends [runtime]System.Object
+  implements class [csLib]StaticsInInterfaces.IGetNext`1<class StaticsTesting/MyRepeatSequence>
+  {
+    .custom instance void [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::.ctor(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags) = ( 01 00 03 00 00 00 00 00 ) 
+    .method public specialname rtspecialname 
+     instance void  .ctor() cil managed
+    {
+      
+      .maxstack  8
+      IL_0000:  ldarg.0
+      IL_0001:  callvirt   instance void [runtime]System.Object::.ctor()
+      IL_0006:  ldarg.0
+      IL_0007:  pop
+      IL_0008:  ret
+    } 
+
+    .method public hidebysig static class StaticsTesting/MyRepeatSequence 
+     'StaticsInInterfaces.IGetNext<StaticsTesting.MyRepeatSequence>.Next'(class StaticsTesting/MyRepeatSequence other) cil managed
+    {
+      .override  method !0 class [csLib]StaticsInInterfaces.IGetNext`1<class StaticsTesting/MyRepeatSequence>::Next(!0)
+      
+      .maxstack  8
+      IL_0000:  ldarg.0
+      IL_0001:  ret
+    } 
+
+  } 
+
+}
+        """]
+    
+#if !NETCOREAPP
+    [<Fact(Skip = "NET472 is unsupported runtime for this kind of test.")>]
+#else
+    [<Fact>]
+#endif
+    let ``F# can implement static methods declared in interfaces from C#`` () =
+
+        let csharpLib = csharpBaseClass 
+
+        let fsharpSource =
+            """
+open System
+open StaticsInInterfaces
+
+type MyRepeatSequence() =
+    interface IGetNext<MyRepeatSequence> with
+        static member Next(other: MyRepeatSequence) : MyRepeatSequence = other 
+
+[<EntryPoint>]
+let main _ =
+
+    let mutable str = MyRepeatSequence ()
+    let res = [ for i in 0..10 do
+                    yield string(str)
+                    str <- MyRepeatSequence.Next(str) ]
+
+    if res <> ["A"; "AA"; "AAA"; "AAAA"; "AAAAA"; "AAAAAA"; "AAAAAAA"; "AAAAAAAA"; "AAAAAAAAA"; "AAAAAAAAAA"; "AAAAAAAAAAA"] then
```

Comment:
Separately we should consider operators.  Consider type variables constrained by `'T when 'T : IAdditionOperator<'T>` sort of thing when `IAdditionOperator<'T>` has an `op_Addition` method.  Then does `'T` satisfy the SRTP constraint implied by F# `+`?

Reply:
@dsyme, if this doesn't clash (with `'T` implementing `op_Addition` explicitly or through yet another interface), I'd say yes, the `+` constraint may be satisfied.

This prompts then, how/where to disambiguate, if there is a clash with several implementations of the `op_Addition` static member?

Reply:
This has now been added to the RFC and implemented